src/main/java/com/checkmarx/jenkins/CheckmarxScanResultsAction.java

```diff
@@ -47,11 +54,18 @@ public ResultsSummary getResultsSummary() {
         for (Object artifact : run.getArtifacts()) {
             if (artifact instanceof Run.Artifact && ((Run.Artifact) artifact).getFileName().contains(PluginUtils.CHECKMARX_AST_RESULTS_JSON)) {
                 try {
-                    byte[] encoded = Files.readAllBytes(Paths.get(((Run.Artifact) artifact).getFile().getCanonicalPath()));
-                    String json = new String(encoded, Charset.defaultCharset());
+                    String artifactHref = ((Run.Artifact) artifact).getHref();
+                    String serverUrl = Jenkins.get().getRootUrl();
+                    String fullUrl = serverUrl + run.getUrl() + "artifact/" + artifactHref;
+                    OkHttpClient client = new ProxyHttpClient().getHttpClient(PluginUtils.getProxy(), 10000, 10000);
+                    Request request = new Request.Builder().url(fullUrl).build();
+                    Response response = client.newCall(request).execute();
+                    ResponseBody responseBody = response.body();
+                    InputStream stream = responseBody.byteStream();
+                    String json = IOUtils.toString(stream);
                     ObjectMapper objectMapper = new ObjectMapper();
                     return objectMapper.readValue(json, ResultsSummary.class);
-                } catch (IOException e) {
+                } catch (IOException | URISyntaxException | CheckmarxException e) {
```

Comment:
Is this cover all the exception that can be returned from this code?

Reply:
no, 
These are only the exceptions I must catch from the methods I used, I will change it to a global exception